src/test/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProviderTest.java

```diff
@@ -0,0 +1,56 @@
+package hudson.plugins.active_directory;
+
+import org.junit.Test;
+
+import static hudson.plugins.active_directory.ActiveDirectoryUnixAuthenticationProvider.escapeCharactersInDistinguisedName;
+import static junit.framework.Assert.assertEquals;
+
+public class ActiveDirectoryUnixAuthenticationProviderTest {
+
+    @Test
+    public void testEscapeCharactersInDistinguisedName() {
+        //Check escape of comma character
+        String distinguisedName = "CN=name,surname,OU=Jenkins-core-developer,DC=domain,DC=com";
+        assertEquals("CN=name\\,surname,OU=Jenkins-core-developer,DC=domain,DC=com", escapeCharactersInDistinguisedName(distinguisedName, false));
+
+        //Check escape of backslash character
+        distinguisedName = "CN=name\\ surname,OU=Jenkins-core-developer,DC=domain,DC=com";
+        assertEquals("CN=name\\\\\\ surname,OU=Jenkins-core-developer,DC=domain,DC=com", escapeCharactersInDistinguisedName(distinguisedName, false));
```

Comment:
this is the wrong result.  
should be
`CN=name\\\\ surname,OU=Jenkins-core-developer,DC=domain,DC=com`